src/dotty/tools/dotc/core/MemberBinding.scala

```diff
@@ -0,0 +1,34 @@
+package dotty.tools.dotc.core
+
+import Names.Name
+import Contexts.Context
+import Types.{Type, TypeBounds}
+
+/** A common super trait of Symbol and Refinement.
+ *  Used to capture the attributes of type parameters
+ *  which can be implemented as either symbols or refinements.
+ */
+trait MemberBinding {
+
+  /** Does this binding represent a type parameter?
+   *  Only in that case the rest of the binding's methods are significant.
+   */
+  def isTypeParam(implicit ctx: Context): Boolean
+
+  /** The name of the member */
+  def memberName(implicit ctx: Context): Name
+
+  /** The info of the member */
```

Comment:
I think that a more accurate description would be "The bounds of the type parameter
@pre isTypeParam = true"